addons/account/tests/test_account_move_line_tax_details.py

```diff
@@ -540,6 +540,46 @@ def test_affect_base_amount_5(self):
         )
         self.assertTotalAmounts(invoice, tax_details)
 
+    def test_affect_base_amount_6(self):
+        affecting_tax = self.env['account.tax'].create({
+            'name': 'Affecting',
+            'amount': 42,
+            'amount_type': 'percent',
+            'type_tax_use': 'sale',
+            'include_base_amount': True,
+            'sequence': 0,
+        })
+
+        affected_tax = self.env['account.tax'].create({
+            'name': 'Affected',
+            'amount': 10,
+            'amount_type': 'percent',
+            'type_tax_use': 'sale',
+            'sequence': 1
+        })
+
+        invoice = self.env['account.move'].create({
+            'move_type': 'out_invoice',
+            'partner_id': self.partner_a.id,
+            'invoice_date': '2021-08-01',
+            'invoice_line_ids': [
+                Command.create({
+                    'name': "affecting + affected",
+                    'account_id': self.company_data['default_account_revenue'].id,
+                    'quantity': 1.0,
+                    'price_unit': 100.0,
+                    'tax_ids': (affecting_tax + affected_tax).ids,
+                }),
+            ]
+        })
+
+        invoice.write({'invoice_line_ids': [Command.delete(invoice.invoice_line_ids.id)]})
+        base_lines, tax_lines = self._dispatch_move_lines(invoice)
+        self.assertFalse(base_lines)
+        self.assertFalse(tax_lines)
+        tax_details = self._get_tax_details()
+        self.assertFalse(tax_details)
```

Comment:
I don't understand the link between the commit message and what is being asserted here

Reply:
this bug also happens when removing all the lines from the invoice the journal item for the affected tax remains

Reply:
I suggested this test because it's a minimal and easy way to see the issue.
Commit message steps should be adapted @HANNICHE-Walid

Reply:
I've updated the commit message